charts/monitoring-config/dashboards/app-requests.json

```diff
@@ -453,109 +456,82 @@
-            "fillOpacity": 0,
-            "gradientMode": "hue",
             "hideFrom": {
               "legend": false,
               "tooltip": false,
               "viz": false
             },
-            "lineInterpolation": "smooth",
-            "lineStyle": {
-              "fill": "solid"
-            },
-            "lineWidth": 1,
-            "pointSize": 5,
             "scaleDistribution": {
               "type": "linear"
-            },
-            "showPoints": "always",
-            "spanNulls": true,
-            "stacking": {
-              "group": "A",
-              "mode": "none"
-            },
-            "thresholdsStyle": {
-              "mode": "off"
             }
-          },
-          "mappings": [],
-          "thresholds": {
-            "mode": "absolute",
-            "steps": [
-              {
-                "color": "green",
-                "value": null
-              },
-              {
-                "color": "red",
-                "value": 80
-              }
-            ]
-          },
-          "unit": "none"
+          }
         },
         "overrides": []
       },
       "gridPos": {
-        "h": 13,
+        "h": 9,
         "w": 24,
         "x": 0,
         "y": 20
       },
-      "id": 5,
+      "id": 4,
       "options": {
+        "calculate": false,
+        "cellGap": 1,
+        "color": {
+          "exponent": 0.5,
+          "fill": "dark-orange",
+          "mode": "scheme",
+          "reverse": false,
+          "scale": "exponential",
+          "scheme": "Viridis",
+          "steps": 64
+        },
+        "exemplars": {
+          "color": "rgba(255,0,255,0.7)"
+        },
+        "filterValues": {
+          "le": 1e-9
+        },
         "legend": {
-          "calcs": [
-            "max"
-          ],
-          "displayMode": "table",
-          "placement": "right",
-          "showLegend": true,
-          "width": 300
+          "show": true
+        },
+        "rowsFrame": {
+          "layout": "auto"
         },
         "tooltip": {
-          "mode": "single",
-          "sort": "none"
+          "show": true,
+          "yHistogram": false
+        },
+        "yAxis": {
+          "axisPlacement": "left",
+          "reverse": false
         }
       },
+      "pluginVersion": "9.5.5",
       "targets": [
         {
           "datasource": {
             "type": "prometheus",
             "uid": "prometheus"
           },
           "editorMode": "code",
-          "exemplar": true,
-          "expr": "label_replace(avg without (pod, instance) (http_request_duration_seconds{namespace=\"${namespace}\", job=~\"${app}\", quantile=~\"${quantile}\"}), \"a\", \"$1\", \"job\", \"(.*)\")",
-          "interval": "",
-          "legendFormat": "{{job}} {{controller}} {{action}} {{quantile}}",
+          "expr": "sum by (le) (increase(http_request_duration_seconds_bucket{namespace=\"${namespace}\", job=~\"${app}\"}[$__rate_interval]))",
```

Comment:
In this one, `increase` actually gives us something other than what we want. This is because `increase` gives the difference in value between the first and last non-null samples of its argument (range vector) whereas we need the per-second rate. In other words, we expect our output to be independent of the chosen sliding window (rate interval) but we were erroneously multiplying each output point by the number of seconds in the rate interval.

You can see this e.g. if you plot just one bucket (say `le="+Inf"` i.e. cumulative total request count) for some particular job as a timeseries and compare the values to the request rate graph.
```suggestion
          "expr": "sum by (le) (rate(http_request_duration_seconds_bucket{namespace=\"${namespace}\", job=~\"${app}\"}[$__rate_interval]))",
```

Reply:
Oh wait, I think I've got my wires crossed here — our output points should be durations not counts of requests.

Reply:
Yeah no I think I was thinking ok in the first post. Do check my working though! 😅

Reply:
I think I could argue it both ways... It's these numbers we're talking about (sloppily underlined in red) I think:

<img width="392" alt="image" src="https://github.com/alphagov/govuk-helm-charts/assets/1696784/c96bea89-1193-4d55-8d82-673c2b54ffde">

- `increase` gives "the absolute number of requests in this bit of the heatmap"
- `rate` gives "the number of requests per second in this bit of the heatmap"

Because it's a heatmap, rather than a line graph, the size of the bucket is observable in the visualisation, so `increase` still makes sense in context, I think. That said, I think it's highly debateable which is more intuitive.... 🤔

I think the argument for using `increase` is strongest for the buckets with a small number of samples:

<img width="392" alt="image" src="https://github.com/alphagov/govuk-helm-charts/assets/1696784/9952343f-3a18-4fdf-ba67-f5530f99e46c">

i.e. (with increase) 

> there were 4 requests between 03:02 and 03:03 which had a duration of between 5 and 10 seconds.

vs. (with rate)

> there were 0.067 requests per second between 03:02 and 03:03 which had a duration of between 5 and 10 seconds.

I think the argument for using `rate` is that it's consistent with other visualisations on the dashboard.

I'm honestly not sure which is better. Interested in your thoughts?

Reply:
One other thing to note: even with `increase`, the values are not always integers, because `$__rate_interval` is not always an even multiplier of the sample interval. So you do get confusing datapoints like "there were 2.67 requests between 03:02 and 03:03".

Reply:
Screenshot of it with rate with units added (which do help clear it up a bit):

<img width="391" alt="image" src="https://github.com/alphagov/govuk-helm-charts/assets/1696784/a6f4b98b-b085-4614-bb1c-41cea0567a14">

Reply:
Or with increase, but with the value in the tooltip labeled so it's more clear what it means:

<img width="423" alt="image" src="https://github.com/alphagov/govuk-helm-charts/assets/1696784/bdf4ccd3-3bc5-4862-b75e-a896e5e20691">

Reply:
Ah yeah that's a fair point; I think if we're labelling it clearly like in your last screenshot then it's fine.